src/Sylius/Behat/Context/Setup/OrderContext.php

```diff
@@ -0,0 +1,167 @@
+     * @param SharedStorageInterface $sharedStorage
+     * @param ObjectManager $objectManager
+     */
+    public function __construct(
+        OrderRepositoryInterface $orderRepository,
+        FactoryInterface $orderFactory,
+        OrderShipmentFactoryInterface $orderShipmentFactory,
+        PaymentFactoryInterface $paymentFactory,
+        FactoryInterface $orderItemFactory,
+        OrderItemQuantityModifierInterface $itemQuantityModifier,
+        SharedStorageInterface $sharedStorage,
+        ObjectManager $objectManager
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->orderFactory = $orderFactory;
+        $this->orderShipmentFactory = $orderShipmentFactory;
+        $this->paymentFactory = $paymentFactory;
+        $this->orderItemFactory = $orderItemFactory;
+        $this->itemQuantityModifier = $itemQuantityModifier;
+        $this->sharedStorage = $sharedStorage;
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * @Given the customer :customer placed an order :orderNumber
+     */
+    public function theCustomerPlacedAnOrder(CustomerInterface $customer, $orderNumber)
+    {
+        /** @var OrderInterface $order */
+        $order = $this->orderFactory->createNew();
+
+        $order->setCustomer($customer);
+        $order->setNumber($orderNumber);
+        $order->setChannel($this->sharedStorage->get('channel'));
+        $order->setCurrency($this->sharedStorage->get('currency'));
+
+        $this->sharedStorage->set('order', $order);
+
+        $this->orderRepository->add($order);
+    }
+
+    /**
+     * @Given /^the customer chose ("[^"]+" shipping) (to "[^"]+") with ("[^"]+" payment)$/
+     */
+    public function theCustomerChoseShippingToWithPayment(
+        ShippingMethodInterface $shippingMethod,
+        AddressInterface $address,
+        PaymentMethodInterface $paymentMethod
+    ) {
+        /** @var OrderInterface $order */
+        $order = $this->sharedStorage->get('order');
+
+        $this->orderShipmentFactory->createForOrder($order);
+        $order->getShipments()->first()->setMethod($shippingMethod);
+
+        $payment = $this->paymentFactory->createWithAmountAndCurrency($order->getTotal(), $order->getCurrency());
+        $payment->setMethod($paymentMethod);
+
+        $order->addPayment($payment);
+
+        $order->setShippingAddress($address);
+        $order->setBillingAddress($address);
+
+        $this->objectManager->flush();
+    }
+
+    /**
+     * @Given the customer bought single :product
+     */
+    public function theCustomerBoughtSingle(ProductInterface $product)
+    {
+        /** @var OrderInterface $order */
+        $order = $this->sharedStorage->get('order');
+
+        /** @var OrderItemInterface $item */
+        $item = $this->orderItemFactory->createNew();
+        $item->setVariant($product->getMasterVariant());
+        $item->setUnitPrice($product->getPrice());
```

Comment:
This is ok for now, but in future improvements we should include calculating that price with the pricing calculator.